### DISCUSSION TITLE
Allow plugins to provide multiple buttons onto the menubar

### DISCUSSION THREAD

**Discussion by OItisTang** [swiftbar/SwiftBar#417] [Ideas]
![toolbar on macOS menubar](https://github.com/swiftbar/SwiftBar/assets/3290572/e1f898dd-564d-4288-9a00-48b03a895c9b)

This is the Netease Music app on macOS which has a multiple-buttons toolbar on the menubar.

I hope SwiftBar allows plugins to add buttons directly onto the menubar,
instead of only allow adding actions into menu.

**Comment by melonamin** [comment]
Technically you can make a SwiftBar plugin that will not present a menu and act like a button. The problem is that currently you can have only one menu bar item per plugin, so if you want to recreate what you have on the screenshot:
* Lyrics - one plugin
* Toolbar - one plugin for each control, 4 in total
* Main app - one plugin

So this will be 6 plugins in total, which is ugly but should work! :) 
I will think about adding an ability for plugins to create more than one menubar item, but no promises
